src/v/kafka/server/request_context.h

```diff
@@ -275,6 +267,33 @@ class request_context {
         usage_mgr().add_bytes_sent(response_size - internal_bytes_sent);
     }
 
+    template<typename ResponseType>
+    ss::future<> log_response(ResponseType res) {
```

Comment:
* Why is this async?

Reply:
It is async because `res.data.errored()` is async.

Reply:
Because `res.data.errored()` is async.